File: go/internal/tools/customization.go
package tools

import (
	"context"
	"fmt"

	"github.com/hektyc/unraid-mcp-server/internal/config"
	"github.com/hektyc/unraid-mcp-server/internal/guards"
	"github.com/hektyc/unraid-mcp-server/internal/mcp"
)

func RegistercustomizationTools(s *mcp.Server, cfg *config.Config) {
	s.RegisterTool(&mcp.ToolDef{
		Name:        "customization_themes",
		Description: "Get available themes. Read-only.",
		Query:       `query { customization { themes { id name } } }`,
	})
	s.RegisterTool(&mcp.ToolDef{
		Name:        "customization_locales",
		Description: "Get available locales. Read-only.",
		Query:       `query { customization { locales { id name } } }`,
	})
	s.RegisterTool(&mcp.ToolDef{
		Name:        "customization_set_theme",
		Description: "Set UI theme.",
		Query:       `mutation SetTheme($themeId: String!) { setTheme(themeId: $themeId) { status } }`,
		Params: map[string]string{
			"theme_id": "string", // required=true
		},
	})
}

File: src/tools/customization.ts
import { type ToolDefinition, type ToolResult } from "../core/types.js";

export function getTools(): ToolDefinition[] {
  const make = (n: string, d: string, s: Record<string, unknown>, fn: (p: any, ctx: any) => Promise<ToolResult>): ToolDefinition => ({
    name: n, description: d, inputSchema: s as any, execute: fn,
  });
  return [
    make("customization_themes", "Get available themes. Read-only.", { type:"object", properties:{} },
      async (_p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("query { customization { themes { id name author } } }") })),
    make("customization_locales", "Get available locales. Read-only.", { type:"object", properties:{} },
      async (_p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("query { customization { locales { id name language } } }") })),
    make("customization_set_theme", "Set UI theme.", { type:"object", properties:{ theme_id:{type:"string"} }, required:["theme_id"] },
      async (p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("mutation SetTheme($themeId: String!) { setTheme(themeId: $themeId) { status } }", p) })),
    make("customization_set_locale", "Set locale.", { type:"object", properties:{ locale_id:{type:"string"} }, required:["locale_id"] },
      async (p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("mutation SetLocale($localeId: String!) { setLocale(localeId: $localeId) { status } }", p) })),
    make("customization_sso", "Get SSO config. Read-only.", { type:"object", properties:{} },
      async (_p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("query { customization { sso { enabled provider } } }") })),
    make("customization_display_details", "Get display details. Read-only.", { type:"object", properties:{} },
      async (_p:any, ctx:any) => ({ isSuccess:true, data: await ctx.query("query { display { case theme temperatureDisplay thresholds locale } }") })),
  ];
}
